lib/util.c

```diff
@@ -334,6 +335,52 @@ msp_binary_interval_search(double query, const double *values, size_t n_values)
     return l;
 }
 
+/* This function follows standard semantics of:
+   numpy.searchsorted(..., side='left') and
+   std::lower_bound() from the standard C++ <algorithm> library
+ */
+const double *
+find_first_upper_bound(const double *values, size_t n_values, double query)
+{
+    if (query > values[n_values - 1]) {
+        return values + n_values;
+    }
+    return values + find_first_upper_bound_or_back(values, n_values, query);
+}
+
+#ifndef __OPTIMIZE__
```

Comment:
I don't think we need to bother with conditional compile paths here. I'd rather compile in the old code as part of the test suite (i.e., copy the implementation in the test C file) and do the comparison as part of unit tests.

Reply:
This code is going to need to change with some of my next changes. After that it should look less confusing. Also after I add some explanation as to how this overall lookup technique works, this test will make more sense I think. It's not really testing old code vs new code. In its current state it does kind of look like that.
I'll leave this un-resolved to remind me to circle back to you about it.